swap/src/cli/transport.rs

```diff
@@ -0,0 +1,28 @@
+use crate::network::tor_transport::TorDialOnlyTransport;
+use crate::network::transport::authenticate_and_multiplex;
+use anyhow::Result;
+use libp2p::core::muxing::StreamMuxerBox;
+use libp2p::core::transport::Boxed;
+use libp2p::dns::TokioDnsConfig;
+use libp2p::tcp::TokioTcpConfig;
+use libp2p::{identity, PeerId, Transport};
+
+/// Creates the libp2p transport for the swap CLI.
+///
+/// The CLI's tranport needs the following capabilities:
+/// - Establish TCP connections
+/// - Resolve DNS entries
+/// - Dial onion-addresses through a running Tor daemon (by connecting to the
+///   socks5 port).
+pub fn new(
```

Comment:
I understand that by composing the transports in this way, it means that if `tor_dial_only` returns `MultiaddrNotSupported` it will fallback to `tcp_with_dns`.

If this is true (and if this is the wanted behavior), I would appreciate if we could rephrase the description here.

Reply:
Yep, you understand correctly. Do note that this was already the case behaviour, it was just implemented inside the tor transport instead of achieving it through composition here.